sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -41,213 +43,303 @@
+            # then use my instance logger
+            logger = request.context.setdefault("logger", options.pop("logger", self.logger))
+            if not logger.isEnabledFor(logging.INFO):
+                return
+            try:
+                parsed_url = list(urllib.parse.urlparse(http_request.url))
+                parsed_qp = urllib.parse.parse_qsl(parsed_url[4], keep_blank_values=True)
+                filtered_qp = [(key, self._redact_query_param(key, value)) for key, value in parsed_qp]
+                # 4 is query
+                parsed_url[4] = "&".join(["=".join(part) for part in filtered_qp])
+                redacted_url = urllib.parse.urlunparse(parsed_url)
+
+                multi_record = os.environ.get(HttpLoggingPolicy.MULTI_RECORD_LOG, False)
+
+                if 'logger_attributes' in request.context:
+                    cosmos_logger_attributes = request.context['logger_attributes']
+                    cosmos_logger_attributes['is_request'] = True
+                elif (logger.filters):
+                    return
+                else:
+                    cosmos_logger_attributes = {
+                        'duration': None,
+                        'status_code': None,
+                        'sub_status_code': None,
+                        'verb': http_request.method,
+                        'url': redacted_url,
+                        'database_name': database_name,
+                        'collection_name': collection_name,
+                        'resource_type': resource_type,
+                        'operation_type': operation_type,
+                        'is_request': True}
+
+                client_settings = self._log_client_settings()
+                db_settings = self._log_database_account_settings()
+                if multi_record:
+                    logger.info(client_settings, extra=cosmos_logger_attributes)
+                    logger.info(db_settings, extra=cosmos_logger_attributes)
+                    logger.info("Request URL: %r", redacted_url, extra=cosmos_logger_attributes)
+                    logger.info("Request method: %r", http_request.method, extra=cosmos_logger_attributes)
+                    logger.info("Request headers:", extra=cosmos_logger_attributes)
+                    for header, value in http_request.headers.items():
+                        value = self._redact_header(header, value)
+                        if value and value != HttpLoggingPolicy.REDACTED_PLACEHOLDER:
+                            logger.info("    %r: %r", header, value, extra=cosmos_logger_attributes)
+                    if isinstance(http_request.body, types.GeneratorType):
+                        logger.info("File upload", extra=cosmos_logger_attributes)
+                        return
+                    try:
+                        if isinstance(http_request.body, types.AsyncGeneratorType):
+                            logger.info("File upload", extra=cosmos_logger_attributes)
```

Comment:
Is file upload actually a scenario in this SDK?
This might be here to duplicate the original core implementation - if that's the case, you could potentially simplify by removing things that really aren't going to apply to Cosmos payloads (as the Core policy was naturally written to accommodate any traffic).